Fix Travis CI

# Description
Travis CI fails for all builds. The relevant Travis log part is probably:
```
openjdk version "9.0.4"
OpenJDK Runtime Environment (build 9.0.4+11)
OpenJDK 64-Bit Server VM (build 9.0.4+11, mixed mode)
$ javac -J-Xmx32m -version
javac 9.0.4
3.54s$ mvn -f gson install -DskipTests=true
[INFO] Scanning for projects...
Downloading from google-maven-central: https://maven-central.storage-download.googleapis.com/repos/central/data/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloading from sonatype: https://oss.sonatype.org/content/repositories/releases/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloading from sonatype-apache: https://repository.apache.org/content/repositories/releases/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for com.google.code.gson:gson-parent:2.8.6-SNAPSHOT: Could not transfer artifact org.sonatype.oss:oss-parent:pom:7 from/to google-maven-central (https://maven-central.storage-download.googleapis.com/repos/central/data/): sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target and 'parent.relativePath' points at wrong local POM @ com.google.code.gson:gson-parent:2.8.6-SNAPSHOT, /home/travis/build/google/gson/pom.xml, line 6, column 11
```

It appears it tries all configured repositories and then fails:
1. maven-central.storage-download.googleapis.com
2. oss.sonatype.org (this just redirects to central!)
3. repository.apache.org

It looks like Maven mixes the exception messages, but the actual problem is:
> sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

If we now compare the last [successful build](https://travis-ci.org/google/gson/jobs/541365646) with the [first known failing](https://travis-ci.org/google/gson/jobs/583845328) (there might have been earlier ones through pull requests, but Travis does not show them), we will see:

<table>
    <thead>
        <tr>
            <th>Successful</th>
            <th>Failed</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
<pre>
java version "9.0.1"
Java(TM) SE Runtime Environment (build 9.0.1+11)
</pre>
            </td>
            <td>
<pre>
openjdk version "9.0.4"
OpenJDK Runtime Environment (build 9.0.4+11)
</pre>
            </td>
        </tr>
    </tbody>
</table>

And indeed there is [JDK-8189357](https://bugs.openjdk.java.net/browse/JDK-8189357) which describes that the `cacerts` file for OpenJDK is empty.  
To fix this a different truststore has to be specified, the certificates have to be imported, or you could switch to OpenJDK > 9 (see JEP [319](https://openjdk.java.net/jeps/319)), e.g. 11 which runs the build successfully as well and is also LTS.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: java
 
 jdk:
-  - oraclejdk9
+  - openjdk11
 
 install: mvn -f gson install -DskipTests=true
 script: mvn -f gson test

